frame/staking/src/benchmarking.rs

```diff
@@ -137,6 +141,131 @@ pub fn create_validator_with_nominators<T: Config>(
+			..
+		} = self;
+
+		let src2_node = Node::<T>::from_id(&src2_stash).ok_or("node not found for src stash")?;
+		let weight_of = Staking::<T>::weight_of_fn();
+		ensure!(
+			!src2_node.is_misplaced(&weight_of),
+			"src2_node must be in proper place after rebag"
+		);
+
+		let total_issuance = T::Currency::total_issuance();
+
+		let dest_bag_thresh = T::CurrencyToVote::to_vote(*dest_bag_thresh, total_issuance);
+		let dest_bag_ids = Bag::<T>::get(dest_bag_thresh)
+			.ok_or("destination bag not found")?
+			.iter()
+			.map(|n| n.voter().id.clone())
+			.collect::<Vec<_>>();
+		let expected_dest_ids = vec![dest1_stash.clone(), src2_stash.clone()];
+		ensure!(
+			dest_bag_ids == expected_dest_ids,
+			"destination bag is not in expected state after rebag"
+		);
+
+		let src_bag_thresh = T::CurrencyToVote::to_vote(*src_bag_thresh, total_issuance);
+		let src_bag_ids = Bag::<T>::get(src_bag_thresh)
+			.ok_or("source bag not found")?
+			.iter()
+			.map(|n| n.voter().id.clone())
+			.collect::<Vec<_>>();
+		let expected_src_ids = vec![src1_stash.clone(), src3_stash.clone()];
+		ensure!(src_bag_ids == expected_src_ids, "source bag not in expected state after rebag");
+
+		Ok(())
+	}
+}
+
+// Build `RebagScenario`.
+fn build_rebag_scenario<T: Config>(
+	src_bag_thresh: BalanceOf<T>,
+	dest_bag_thresh: BalanceOf<T>,
+) -> Result<RebagScenario<T>, &'static str> {
+	// The most expensive case for this rebag-ing:
+	//
+	// - The node to be rebagged should exist as a non-terminal node in a bag with at least
+	//   2 other nodes so both its prev and next are nodes that will need be updated
+	//   when it is removed.
+	// - The destination bag is not empty, because then we need to update the `next` pointer
+	//   of the previous node in addition to the work we do otherwise.
+
+	// create_stash_controller takes a factor, so we compute it.
+	let src_factor: BalanceOf<T> =
+		BalanceOf::<T>::from(src_bag_thresh) * 10u32.into() / T::Currency::minimum_balance();
+	let dest_factor: BalanceOf<T> =
+		BalanceOf::<T>::from(dest_bag_thresh) * 10u32.into() / T::Currency::minimum_balance();
+
+	let (dest1_stash, dest1_controller) =
+		create_stash_controller_b::<T>(USER_SEED, dest_factor, Default::default())?;
+	Staking::<T>::validate(RawOrigin::Signed(dest1_controller).into(), Default::default())?;
+
+	let (src1_stash, src1_controller) =
+		create_stash_controller_b::<T>(USER_SEED + 1, src_factor, Default::default())?;
+	Staking::<T>::validate(RawOrigin::Signed(src1_controller.clone()).into(), Default::default())?;
+
+	let (src2_stash, src2_controller) =
+		create_stash_controller_b::<T>(USER_SEED + 2, src_factor, Default::default())?;
+	Staking::<T>::validate(RawOrigin::Signed(src2_controller.clone()).into(), Default::default())?;
+
+	let (src3_stash, src3_controller) =
+		create_stash_controller_b::<T>(USER_SEED + 3, src_factor, Default::default())?;
+	Staking::<T>::validate(RawOrigin::Signed(src3_controller.clone()).into(), Default::default())?;
+
```

Comment:
This stuff is going into a new pallet anyways, so we shoudl think about it from now. I'd just manually inject some data into the Bags/Nodes data structure, instead of trying so hard to do it the "legal" way through staking.